lib/c-api/src/wasm_c_api/externals/function.rs

```diff
@@ -1,27 +1,31 @@
+use super::super::context::wasm_context_t;
 use super::super::store::wasm_store_t;
 use super::super::trap::wasm_trap_t;
 use super::super::types::{wasm_functype_t, wasm_valkind_enum};
 use super::super::value::{wasm_val_inner, wasm_val_t, wasm_val_vec_t};
 use super::CApiExternTag;
+use std::cell::RefCell;
 use std::convert::TryInto;
 use std::ffi::c_void;
 use std::mem::MaybeUninit;
-use std::sync::{Arc, Mutex};
-use wasmer_api::{Function, RuntimeError, Val};
+use std::rc::Rc;
+use wasmer_api::{Function, RuntimeError, Value};
 
 #[derive(Debug, Clone)]
 #[allow(non_camel_case_types)]
 #[repr(C)]
 pub struct wasm_func_t {
     pub(crate) tag: CApiExternTag,
     pub(crate) inner: Box<Function>,
+    pub(crate) context: Option<Rc<RefCell<wasm_context_t>>>,
```

Comment:
Why do we need to add the new `context` field?
`Function` should already have a reference to the store. And we can retrieve the context from that function store... so why duplicate it? (this comment should apply also for `Global`, ...)

Reply:
Because that's the point of the `Context` API, `Function` doesn't have a reference to the store because it's owned by the `Context`/store. `wasmer_api::Function` is just a handle (an integer) to the `context`'s vectors of objects.